src/cron/arweave/mod.rs

```diff
@@ -217,19 +201,17 @@ impl Arweave {
         );
 
         let url = format!("{}/graphql?query={}", self.get_host(), raw_query);
-        let client = ctx.get_client();
         let data = format!(
             "{{\"query\":\"{}\",\"variables\":{}}}",
             raw_query, raw_variables
         );
 
-        let req: http::Request<String> = http::request::Builder::new()
-            .method(http::Method::POST)
-            .uri(http::uri::Uri::from_str(&url).unwrap())
-            .body(data)
-            .unwrap();
-        let req: reqwest::Request = reqwest::Request::try_from(req).unwrap();
+        let client = reqwest::Client::new();
+        let body = serde_json::from_str::<ReqBody>(&data);
+        let req = client.post(&url).json(&body.unwrap()).build().unwrap();
```

Comment:
There's couple of issues here:
- first you create a new `reqwest` client that is only used for creating a request, this is quite expensive thing to do
- secondly, you first deserialize `data` into `ReqBody` and then when passing it to client to build the request, it gets serialized back to string representation. This is very expensive round trip

If you want to make sure your request body is valid, just create `ReqBody` and then serialize it into JSON. Or keep it as string form, that is the cheapest choice. Also, what is wrong with the original way of building the request?

Reply:
The arweave.net api doesn't accept the query as a body argument, so I have to parse it in the query as an URI argument. The URI doesn't accept blank spaces, and formatting as %20 breaks the query on arweave side. This query is weird but is the way I found for it to work. Any suggestions that work are welcome.

Reply:
OK, let's fix this in a follow up PR. I need to test something, but I wasn't able to verify what I wanted in rust playground so I'll write a separate test and fix for this one.